src/lib/queries/get-must.spec.ts

```diff
@@ -14,4 +31,162 @@ describe('getMustQuery', () => {
+
+        expect(result.total).toBeGreaterThan(0)
+        result.documents.forEach(document => expect(document.propertyAreaSquared).toBeDefined())
+    })
+
+    it('should query elasticsearch for must query with range query', async () => {
+        const service = app.get(ElasticsearchService)
+
+        const result = await service.search(HomeDocument, {
+            size: 10,
+            query: getBoolQuery({
+                ...getMustQuery(getExistsQuery('propertyAreaSquared')),
+                ...getMustQuery(
+                    getRangeQuery('propertyAreaSquared', {
+                        gte: 500000,
+                        lte: 900000
+                    })
+                )
+            })
+        })
+
+        expect(result.total).toBeGreaterThan(0)
+        result.documents.forEach(document => expect(document.propertyAreaSquared).toBeGreaterThanOrEqual(500000))
+        result.documents.forEach(document => expect(document.propertyAreaSquared).toBeLessThanOrEqual(900000))
+    })
+
+    it('should query elasticsearch for nested must query ', async () => {
+        const service = app.get(ElasticsearchService)
+
+        const result = await service.search(HomeDocument, {
+            size: 10,
+            query: getBoolQuery(
+                getMustQuery([
+                    getTermQuery('hasProperty', true),
+                    getBoolQuery({
+                        ...getMustQuery({
+                            ...getTermQuery('propertyType.keyword', PropertyType.Flat)
+                        })
+                    })
+                ])
+            )
+        })
+
+        expect(result.total).toBeGreaterThan(0)
+        result.documents.forEach(document => {
+            expect(document.hasProperty).toBe(true)
+            expect(document.propertyType).toBe(PropertyType.Flat)
+        })
+    })
+
+    it('should query elasticsearch for different queries in must query', async () => {
+        const service = app.get(ElasticsearchService)
+
+        const query = 'c'
+        const result = await service.search(HomeDocument, {
+            size: 10,
+            query: getBoolQuery(
+                getMustQuery([
+                    getExistsQuery('propertyAreaSquared'),
+                    getTermQuery('propertyType.keyword', PropertyType.Flat),
+                    getRangeQuery('propertyAreaSquared', { lte: 900000 }),
+                    getMatchPhrasePrefixQuery('address', query)
+                ])
+            )
+        })
+
+        expect(result.total).toBeGreaterThan(0)
+        result.documents.forEach(document => {
+            expect(document.propertyAreaSquared).toBeDefined()
+            expect(document.propertyType).toBe(PropertyType.Flat)
+            expect(document.propertyAreaSquared).toBeLessThanOrEqual(900000)
+
+            const words = document.address.toLowerCase().split(' ')
+            const hasWordStartingWithQuery = words.some(word => word.startsWith(query.toLowerCase()))
```

Comment:
shouldn't we replace `startsWith` with `includes`? I think if we regenerate the seed - the results might not start with the query `c`

Reply:
hmm but I'm using `getMatchPhrasePrefixQuery`, it should `startsWith`